lib/rules/no-empty-function.js

```diff
@@ -83,13 +103,28 @@ function getKind(node) {
 	return prefix + kind[0].toUpperCase() + kind.slice(1);
 }
 
+/**
+ * Checks if a constructor function has parameter properties.
+ * @param {ASTNode} node The function node to examine.
+ * @returns {boolean} True if the constructor has parameter properties, false otherwise.
+ */
+function isParameterPropertiesConstructor(node) {
+	return (
+		node.parent.type === "MethodDefinition" &&
+		node.parent.kind === "constructor" &&
```

Comment:
No unit tests fail if these are removed. Either this is unnecessary code or some testing is missing:

```suggestion
```

`TSParameterProperty` nodes can only appear in constructor functions, so I'm pretty sure it's just unnecessary code.